src/modules/Game.class.js

```diff
@@ -50,19 +99,170 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    return this.#status;
+  }
 
   /**
-   * Starts the game.
+   * Update game status.
+   *
+   * @param {string} value
+   *
    */
-  start() {}
+  setStatus(value) {
+    this.#status = value;
+  }
 
   /**
-   * Resets the game.
+   * This method shoud be private, but the tests consider it as public
    */
-  restart() {}
+  start() {
+    this.#score = 0;
+    this.#status = 'playing';
+    this.#state = this.#getStateCopy(this.#initialState);
+
+    // Is called twice by requirements
+    this.#randomizer();
+    this.#randomizer();
+  }
+
+  // Resets the game.
+  restart() {
+    this.#score = 0;
+    this.#status = 'playing';
+    this.#state = this.#getStateCopy(this.#initialState);
+    this.#status = 'idle';
+  }
 
-  // Add your own methods here
+  /**
+   * Put 2 or 4 in freely cells if they exists
+   */
+  #randomizer() {
+    const empties = [];
+    let idxSorted;
+
+    for (let line = 0; line < this.#SIZE; line++) {
+      for (let column = 0; column < this.#SIZE; column++) {
+        if (this.#state[line][column] === 0) {
+          empties.push([line, column]);
+        }
+      }
+    }
+
+    if (empties.length > 0) {
+      idxSorted = Math.floor(Math.random() * empties.length);
+
+      this.#state[empties[idxSorted][0]][empties[idxSorted][1]] =
+        Math.floor(Math.random() * 10) < 9 ? 2 : 4;
+    }
+  }
+
+  // Matrix transformation transpose operation.
+  #tranpose() {
+    let tempRetriveVar;
+
+    for (let ln = 0; ln < this.#SIZE; ln++) {
+      for (let cl = ln; cl < this.#SIZE; cl++) {
+        tempRetriveVar = this.#state[ln][cl];
+        this.#state[ln][cl] = this.#state[cl][ln];
+        this.#state[cl][ln] = tempRetriveVar;
+      }
+    }
+  }
+
+  // Only reverte this.#state array lines, it dont to revert any array.
+  #reverse() {
+    this.#state.forEach((e, i, arr) => arr[i].reverse());
+  }
+
+  // Check if movements isn't possible than change this.status.
+  #checkIfLose() {
+    let allSides = [];
+
+    for (let column = 0; column < this.#SIZE; column++) {
+      for (let line = 0; line < this.#SIZE; line++) {
+        if (this.#state[line][column] !== 0) {
+          allSides.push(
+            this.#state[line - 1] !== undefined
+              ? this.#state[line - 1][column]
+              : undefined,
```

Comment:
May use the [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) instead of ternary `this.#state[line - 1]?.[column]`